Simple integration with Ramp Network

gkapkowski opened the pull request:
![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/114996/73456699-9eeebc00-4372-11ea-80b9-fde12d215d43.gif)

(ignore the widget position and size, I have tiling window manger)

chaitanyapotti commented:
Hi, 
we provide dapps to invoke the payment API directly from the [torus-embed](https://github.com/torusresearch/torus-embed) using this [API](https://docs.tor.us/api-reference/topup)
The current partners for this API are moonpay, wyre and coindirect. 
If you wish to be integrated here, there are certain requirements we have.
- Not use a sdk
- Provide a quote API to get a quote
- Have a hosted website which will be shown in popup triggered by torus
- Allow form inputs in the hosted website to be picked up from query params of the url
- Have a redirect url support to redirect to after the payment flow is completed
- Ping an endpoint of torus to notify about the tx details + status(webhook)

iamonuwa commented on app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue:
```diff
@@ -0,0 +1,35 @@
+export default {
+  methods: {
+    showWidget() {
+      console.log(arguments)
```
Remove this console

gkapkowski replied:
Done

iamonuwa commented on app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue:
```diff
@@ -0,0 +1,35 @@
+        hostLogoUrl: 'https://tor.us/assets/img/torus-logo.svg',
+        userAddress: this.$store.state.selectedAddress
+      })
+        //.on('*', event => console.log(event))
```
Remove this if not in use

gkapkowski replied:
Done

gkapkowski commented:
@chaitanyapotti I've updated the integration with our API call. Can you tell me more about calling you webhook?

iamonuwa commented:
@gkapkowski just asking though, your logo on our dark theme looks off abit. Do you think you can fix that?

chaitanyapotti commented:
We will create an endpoint for you. (rampnetwork-api.tor.us/transaction)
Please make a POST request to that endpoint with the tx object.
Include a signature created from tx object as a header in the POST request which we can verify with a secret API Key which is known to both of us.


Furthermore, please merge current develop of torus into your branch

Thanks,

gkapkowski commented:
@iamonuwa I've fixed the logo. Thanks for pointing this out!

@chaitanyapotti I've added webhook configuration to the widget. The description and examples of tx object can be found in [Webhooks Guide](https://docs-instant.ramp.network/guide/events/webhooks/) and [Webhooks Reference](https://docs-instant.ramp.network/reference/webhooks/), We will add signature header this week. How will we receive secret?

chaitanyapotti commented:
@gkapkowski  The secret would be an API Key generated by you and shared with us.

gkapkowski commented:
@chaitanyapotti we decided to implement signatures based on ECDSA (the same as ethereum signatures) so that we can share our public key in documentation and the verification code on your side can use it. Will that work for you?
It's much easier than sharing secrets and you already work with libraries that have this verification mechanism implemented.

chaitanyapotti commented:
That would totally work.
Thanks

chaitanyapotti commented:
@gkapkowski 
We are ready for the merge as the backend is ready. Can you rebase your repo with the current develop branch?

gkapkowski commented:
@chaitanyapotti I did rebase, please verify.
